analyses/cms-open-data-ttbar/ttbar_analysis_pipeline.py

```diff
@@ -448,107 +412,92 @@ def get_query(source: ObjectStream) -> ObjectStream:
+
+print(f"processes in fileset: {list(fileset.keys())}")
+print(f"\nexample of information in fileset:\n{{\n  'files': [{fileset['ttbar__nominal']['files'][0]}, ...],")
+print(f"  'metadata': {fileset['ttbar__nominal']['metadata']}\n}}")
+
+# %% [markdown]
+# ### Caching the queried datasets with `ServiceX`
+#
+# Using the queries created with `func_adl`, we are using `ServiceX` to read the ATLAS Open Data files to build cached files with only the specific event information as dictated by the query.
 
-    import inspect
- 
-    query_string = inspect.getsource(get_query).split("return source.")[-1]  # extract query from function defined previously
+# %%
+if USE_SERVICEX:
     
-    sample_names = ["ttbar__nominal", "ttbar__scaledown", "ttbar__scaleup", "ttbar__ME_var", "ttbar__PS_var",
-                    "single_top_s_chan__nominal", "single_top_t_chan__nominal", "single_top_tW__nominal", "wjets__nominal"]
-    sample_names = ["single_top_s_chan__nominal"]  # for quick tests: small dataset with only 50 files
-    sample_list = []
-
-    for sample_name in sample_names:
-        sample_list.append({"Name": sample_name, "RucioDID": f"user.ivukotic:user.ivukotic.{sample_name}", "Tree": "events", "FuncADL": query_string})
-
-
-    databinder_config = {
-                            "General": {
-                                           "ServiceXBackendName": "uproot",
-                                            "OutputDirectory": "outputs_databinder",
-                                            "OutputFormat": "root",
-                                            "IgnoreServiceXCache": SERVICEX_IGNORE_CACHE
-                            },
-                            "Sample": sample_list
-                        }
-
-    sx_db = DataBinder(databinder_config)
-    # out = sx_db.deliver(timer=True)
-    parquet_paths = sx_db._sx.get_servicex_data()  # only run transform, do not download as well
+    # dummy dataset on which to generate the query
+    dummy_ds = ServiceXSourceUpROOT("cernopendata://dummy", "events", backend_name="uproot")
+
+    # tell low-level infrastructure not to contact ServiceX yet, only to
+    # return the qastle string it would have sent
+    dummy_ds.return_qastle = True
+
+    # create the query
+    query = get_query(dummy_ds)
+    query = query.value()
+
+    # now we query the files and create a fileset dictionary containing the
+    # URLs pointing to the queried files
+
+    t0 = time.time()
+
+    fileset_queried = {}
+
+    for process in fileset.keys():
+        ds = ServiceXDataset(fileset[process]['files'], backend_name="uproot", ignore_cache=SERVICEX_IGNORE_CACHE)
+        files = ds.get_data_rootfiles_uri(query, as_signed_url=True)
+
+        fileset_queried[process] = {"files": [f.url for f in files],
+                                    "metadata": fileset[process]['metadata']
+                                   }
```

Comment:
I missed this in #96, but is there any reason we cannot directly override `fileset_queried[process]["files"]` and thereby avoid having to worry about the metadata? That should leave whatever else is in the dictionary unchanged (in case we adjust that in the future).

Reply:
I missed that `fileset_queried` is a new dictionary, what I had in mind was overriding `fileset` since that happens a few cells down anyway.

Reply:
Fixed in [9587d6c](https://github.com/iris-hep/analysis-grand-challenge/pull/103/commits/9587d6c0f7766965c0f95282c5a78fbcf1c14001)